std/stdio.d

```diff
@@ -741,10 +742,13 @@ $(D rawRead) always reads in binary mode on Windows.
                 scope(exit) __fhnd_info[fd] = info;
             }
         }
-        immutable result =
+        immutable fread_result =
             fread(buffer.ptr, T.sizeof, buffer.length, _p.handle);
+        immutable fread_success = (fread_result == buffer.length);
+        if (fread_success)
+            return buffer;
         errnoEnforce(!error);
-        return result ? buffer[0 .. result] : null;
+        return buffer[0 .. fread_result];
```

Comment:
Actually, I read the docs for fread, error only occurs on a short read, so this part of the change makes sense. Disregard my recommendation to use the .ptr member.